main/src/library/Reducible.flix

```diff
@@ -0,0 +1,183 @@
+/*
+ *  Copyright 2022 Jakob Schneider Villumsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+///
+/// A type class for types that can be reduced to a summary value.
+///
+/// `Reducible` is like a non-empty `Foldable` and may only be implemented on non-empty data structures.
+///
+/// This type class is heavily inspired by the Cats implementation of Reducible:
```

Comment:
I am all for giving credit (and we try to in various places), but I would not do it here in the documentation.